webapps-schema/src/main/java/io/camunda/webapps/schema/entities/usermanagement/EntityJoinRelation.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.webapps.schema.entities.usermanagement;
+
+public class EntityJoinRelation {
+
+  private String name;
+  private Long parent;
+
+  public String getName() {
+    return name;
+  }
+
+  public EntityJoinRelation setName(final String name) {
+    this.name = name;
+    return this;
+  }
+
+  public Long getParent() {
+    return parent;
+  }
+
+  public EntityJoinRelation setParent(final Long parent) {
+    this.parent = parent;
+    return this;
+  }
+
+  public enum IdentityJoinRelationshipType {
+    GROUP("group"),
+    MEMBER("member");
```

Comment:
I think it'd be nice to move this so that we don't need to pack all types into one global enum.
For example, we could make it a nested enum in GroupIndex and either reference entries by name or parametrize EntityJoinRelation:
```java
class GroupIndex {
  …
  enum JoinName {
    GROUP,
    MEMBER
  }
  …
}

class EntityJoinRelation<E extends Enum<E>> {
  E getName() {…}
}
```
Another option (and I think I like this best) would be a factory for EntityJoinRelation and no enum at all:
```java
class GroupIndex {
  …
  public static final joinRelationFactory = new EntityJoinRelationFactory("group", "member");
  …
}

// and then:
EntityJoinRelation r = GroupIndex.joinRelationFactory.createParent();
EntityJoinRelation r = GroupIndex.joinRelationFactory.createChild(parentId);
```

Reply:
@emulbreh with the current approach we can just expand the enum with `ROLE` and `TENANT` values and reuse the class in the Role- and Tenant-related handlers as well. 

❓ Why is it a problem to pack multiple values in a single enum?

Reply:
> Why is it a problem to pack multiple values in a single enum?

It doesn't help to ensure that the right enum values are used for an index. 
We could leave it untyped and just work with the name strings, but then I'd remove the enum entirely and just keep constants in the handler or index descriptor.